File: 351003/Kisel/Rest/Task310RestApi/Task310RestApi/DTOs/Request/CreatorRequestTo.cs
using System.ComponentModel.DataAnnotations;

namespace Task310RestApi.DTOs.Request
{
    public class CreatorRequestTo
    {
        [Required(ErrorMessage = "Login is required")]
        [StringLength(64, MinimumLength = 2, ErrorMessage = "Login must be between 2 and 64 characters")]
        public required string Login { get; set; }

        [Required(ErrorMessage = "Password is required")]
        [StringLength(128, MinimumLength = 8, ErrorMessage = "Password must be between 8 and 128 characters")]
        public required string Password { get; set; }

        [Required(ErrorMessage = "Firstname is required")]
        [StringLength(64, MinimumLength = 2, ErrorMessage = "Firstname must be between 2 and 64 characters")]
        public required string Firstname { get; set; }

        [Required(ErrorMessage = "Lastname is required")]
        [StringLength(64, MinimumLength = 2, ErrorMessage = "Lastname must be between 2 and 64 characters")]
        public required string Lastname { get; set; }
    }
}

File: 351004/Shkredova/task310-rest/src/main/java/com/example/task310/dto/CreatorRequestTo.java
package com.example.task310.dto;

import jakarta.validation.constraints.Size;
import jakarta.validation.constraints.NotBlank;

public class CreatorRequestTo {

    @NotBlank(message = "Login is required")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String login;

    @NotBlank(message = "Password is required")
    @Size(min = 8, max = 128, message = "Password must be between 8 and 128 characters")
    private String password;

    @NotBlank(message = "Firstname is required")
    @Size(min = 2, max = 64, message = "Firstname must be between 2 and 64 characters")
    private String firstname;

    @NotBlank(message = "Lastname is required")
    @Size(min = 2, max = 64, message = "Lastname must be between 2 and 64 characters")
    private String lastname;

    // Геттеры и сеттеры
    public String getLogin() { return login; }
    public void setLogin(String login) { this.login = login; }

    public String getPassword() { return password; }
    public void setPassword(String password) { this.password = password; }

    public String getFirstname() { return firstname; }
    public void setFirstname(String firstname) { this.firstname = firstname; }

    public String getLastname() { return lastname; }
    public void setLastname(String lastname) { this.lastname = lastname; }
}